sdk/src/main/scala/com/horizen/account/api/http/AccountWalletApiRoute.scala

```diff
@@ -0,0 +1,152 @@
+    withAuth {
+      entity(as[ReqCreateKey]) { _ =>
+        withNodeView { sidechainNodeView =>
+          val wallet = sidechainNodeView.getNodeWallet
+          val secret = PrivateKeySecp256k1Creator.getInstance().generateNextSecret(wallet)
+          val future = sidechainNodeViewHolderRef ? AbstractSidechainNodeViewHolder.ReceivableMessages.LocallyGeneratedSecret(secret)
+          Await.result(future, timeout.duration).asInstanceOf[Try[Unit]] match {
+            case Success(_) =>
+              ApiResponseUtil.toResponse(RespCreatePrivateKeySecp256k1(secret.publicImage()))
+            case Failure(e) =>
+              ApiResponseUtil.toResponse(ErrorSecretNotAdded("Failed to create key pair.", JOptional.of(e)))
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Check balance of the account.
+   * Return 0 if account doesn't exist.
+   */
+  def getBalance: Route = (post & path("getBalance")) {
+    withAuth {
+      entity(as[ReqGetBalance]) { body =>
+        applyOnNodeView { sidechainNodeView =>
+          try {
+            val fromAddr = new AddressProposition(BytesUtils.fromHexString(body.address))
+            val fromBalance = sidechainNodeView.getNodeState.getBalance(fromAddr.address())
+            ApiResponseUtil.toResponse(RespGetBalance(fromBalance))
+          }
+          catch {
+            case e: Exception =>
+              ApiResponseUtil.toResponse(ErrorCouldNotGetBalance("Could not get balance", JOptional.of(e)))
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * Check total balance of the wallet.
+   */
+  def getTotalBalance: Route = (post & path("getTotalBalance")) {
+    withAuth {
+      entity(as[ReqGetTotalBalance]) { _ =>
+        // TODO add an argument for listing also all addresses balance
+        applyOnNodeView { sidechainNodeView =>
+          try {
+            val wallet = sidechainNodeView.getNodeWallet
+            val addressList = wallet.secretsOfType(classOf[PrivateKeySecp256k1])
+            if (addressList.isEmpty) {
+              return ApiResponseUtil.toResponse(RespGetBalance(BigInteger.ZERO))
+            }
+
+            val listOfAddressPropositions = addressList.asScala.map(s =>
+              s.publicImage().asInstanceOf[AddressProposition])
+            var totalBalance = BigInteger.ZERO
+            listOfAddressPropositions.foreach(address =>
+              totalBalance = totalBalance.add(sidechainNodeView.getNodeState.getBalance(address.address())))
+
+            ApiResponseUtil.toResponse(RespGetBalance(totalBalance))
+          }
+          catch {
+            case e: Exception =>
+              ApiResponseUtil.toResponse(ErrorCouldNotGetBalance("Could not get balance", JOptional.of(e)))
+          }
+        }
+      }
+    }
+  }
+
+}
+
+object AccountWalletRestScheme {
+  @JsonView(Array(classOf[Views.Default]))
+  private[api] case class RespCreatePrivateKeySecp256k1(proposition: Proposition) extends SuccessResponse
```

Comment:
Consider using the `RespCreatePrivateKey` case class in UTXO model, which could be moved in the base class